File: firebase-api-library/src/main/java/com/fazemeright/firebase_api_library/api/Storable.java
package com.fazemeright.firebase_api_library.api;

import java.util.Map;
import javax.annotation.Nonnull;

public interface Storable {

  /**
   * Get map of the object
   *
   * @return map of data
   */
  @Nonnull
  Map<String, Object> getHashMap();

  /**
   * ID
   *
   * @return id
   */
  long getId();
}

File: firebase-api-library/src/main/java/com/fazemeright/library/api/Storable.kt
package com.fazemeright.library.api

interface Storable {
    /**
     * Get map of the object.
     *
     * @return map of data
     */
    val hashMap: Map<String, Any>

    /**
     * ID.
     *
     * @return id
     */
    val id: Long
}